esp-box/esp-box.c

```diff
@@ -323,7 +343,9 @@ static lv_disp_t *lvgl_port_display_init(void)
 static esp_err_t lvgl_port_indev_init(void)
 {
     static lv_indev_drv_t indev_drv_tp;
+    static lv_indev_drv_t indev_drv_btn;
```

Comment:
How is the user supposed to assign this button to specific coordinates on the screen, if the handler is not exposed to him?

See docs https://docs.lvgl.io/latest/en/html/porting/indev.html#button

Reply:
> How is the user supposed to assign this button to specific coordinates on the screen, if the handler is not exposed to him?
> 
> See docs https://docs.lvgl.io/latest/en/html/porting/indev.html#button
> 
> and mainly this function [lv_indev_set_button_points](https://docs.lvgl.io/8.3/overview/indev.html?highlight=lv_indev_set_button_points#_CPPv426lv_indev_set_button_pointsP10lv_indev_tA_K10lv_point_t)

yes, it't use 'lv_indev_set_button_points' to virtual a fixed coordinate area
`lv_obj_get_click_area(btn_return, &a);lv_indev_set_button_points(ui_get_button_indev(), points_array);`

Reply:
@espressif2022 Please, where can I find the function `ui_get_button_indev()`. Have you got any working example please?

Reply:
> 

You can refer to this [file](https://github.com/espressif/esp-box/blob/master/examples/factory_demo/main/gui/ui_about_us.c)
It find the handle register by `lv_indev_drv_init`, here is `indev_drv_btn`. 
This is not a standard TP, the general TP won't have this button

Reply:
I still don't see any chance, how to get `indev_drv_btn` for use it in BSP example. There is missing simillar function like `ui_get_button_indev()` in BSP.

Reply:
> I still don't see any chance, how to get `indev_drv_btn` for use it in BSP example. There is missing simillar function like `ui_get_button_indev()` in BSP.

I'm a bit confused that what we need to do here about this button event?
BSP reads the button event, how to handle this event is upto the application?

Reply:
Application missing pointer to `indev_drv_btn`. Which is needed by `lv_indev_set_button_points(...)`.

Reply:
> Application missing pointer to `indev_drv_btn`. Which is needed by `lv_indev_set_button_points(...)`.

If `indev_drv_btn` is null, application cann't execute `lv_indev_set_button_points`, home button is useless.